src/main/java/com/radixdlt/crypto/ECDSASignature.java

```diff
@@ -126,4 +131,19 @@ public int hashCode() {
 	public SignatureScheme signatureScheme() {
 		return SignatureScheme.ECDSA;
 	}
+
+	// ### From Client library ###
+	public static ECDSASignature decodeFromDER(byte[] bytes) {
+		DLSequence seq;
+		ASN1Integer r, s;
+		try (ASN1InputStream decoder = new ASN1InputStream(bytes)) {
+			seq = (DLSequence) decoder.readObject();
+			r = (ASN1Integer) seq.getObjectAt(0);
+			s = (ASN1Integer) seq.getObjectAt(1);
+		} catch (ClassCastException | IOException e) {
+			throw new IllegalArgumentException(e);

```

Comment:
Should add an explanatory message to this exception

Reply:
good point!